File: semana_6/py/gestion_productos.py
class Producto:
    def __init__(self, nombre, precio, categoria):
        self.nombre = nombre
        self.precio = precio
        self.categoria = categoria

    def mostrar_detalle(self):
        print(f"Nombre: {self.nombre}\nPrecio: {self.precio}\nCategoría: {self.categoria}")

class Electronico(Producto):
    def __init__(self, nombre, precio, marca):
        super().__init__(nombre, precio, "Electrónico")
        self.marca = marca

    def mostrar_detalle(self):
        super().mostrar_detalle()
        print(f"Marca: {self.marca}")

class Alimenticio(Producto):
    def __init__(self, nombre, precio, fecha_caducidad):
        super().__init__(nombre, precio, "Alimenticio")
        self.fecha_caducidad = fecha_caducidad

    def mostrar_detalle(self):
        super().mostrar_detalle()
        print(f"Fecha de caducidad: {self.fecha_caducidad}")

class Vestimenta(Producto):
    def __init__(self, nombre, precio, talla):
        super().__init__(nombre, precio, "Vestimenta")
        self.talla = talla

    def mostrar_detalle(self):
        super().mostrar_detalle()
        print(f"Talla: {self.talla}")

laptop = Electronico("Laptop", 1200.0, "XYZ")
laptop.mostrar_detalle()

manzana = Alimenticio("Manzana", 2.5, "2023-10-01")
manzana.mostrar_detalle()

camisa = Vestimenta("Camisa", 30.0, "M")
camisa.mostrar_detalle()

File: semana_6/c++/gestion_productos.cc
#include <iostream>
#include <string>

class Producto {
public:
    std::string nombre;
    double precio;
    std::string categoria;

    Producto(std::string nombre, double precio, std::string categoria)
        : nombre(nombre), precio(precio), categoria(categoria) {}

    virtual void mostrar_detalle() {
        std::cout << "Nombre: " << nombre << "\nPrecio: " << precio << "\nCategoria: " << categoria << "\n";
    }
};

class Electronico : public Producto {
public:
    std::string marca;

    Electronico(std::string nombre, double precio, std::string marca)
        : Producto(nombre, precio, "Electronico"), marca(marca) {}

    void mostrar_detalle() override {
        Producto::mostrar_detalle();
        std::cout << "Marca: " << marca << "\n";
    }
};

class Alimenticio : public Producto {
public:
    std::string fechaCaducidad;

    Alimenticio(std::string nombre, double precio, std::string fechaCaducidad)
        : Producto(nombre, precio, "Alimenticio"), fechaCaducidad(fechaCaducidad) {}

    void mostrar_detalle() override {
        Producto::mostrar_detalle();
        std::cout << "Fecha de caducidad: " << fechaCaducidad << "\n";
    }
};

class Vestimenta : public Producto {
public:
    std::string talla;

    Vestimenta(std::string nombre, double precio, std::string talla)
        : Producto(nombre, precio, "Vestimenta"), talla(talla) {}

    void mostrar_detalle() override {
        Producto::mostrar_detalle();
        std::cout << "Talla: " << talla << "\n";
    }
};

int main() {
    Electronico laptop("Laptop", 1200.0, "XYZ");
    laptop.mostrar_detalle();

    Alimenticio fruta("Manzana", 2.5, "2023-10-01");
    fruta.mostrar_detalle();

    Vestimenta camisa("Camisa", 30.0, "M");
    camisa.mostrar_detalle();

    return 0;
}
